prometheus_es_exporter/__init__.py

```diff
@@ -72,6 +72,13 @@ def group_metrics(metrics):
 def update_gauges(metrics):
     metric_dict = group_metrics(metrics)
 
+    query_prefix = metrics[0][0][0]
+    query_metric_names = set()
+    query_metrics_found = set()
+    for metric_name in gauges.keys():
+        if metric_name.startswith(query_prefix):
```

Comment:
This check might not be 100% foolproof if one query name is contained in another query name. No idea how to make this perfect at the moment.